test-utils/src/transactions.rs

```diff
@@ -215,14 +218,29 @@ impl<R: Rng + CryptoRng> TransactionsGenerator<R> {
         if matches!(sender, OutgoingType::DeleteValidator) {
             value = Coin::from_u64_unchecked(Policy::VALIDATOR_DEPOSIT)
                 .checked_sub(fee)
-                .expect("Fee should not be higher than validator deposit");
+                .expect("Fee must not be higher than validator deposit");
+            if fail_sender {
+                value += Coin::from_u64_unchecked(1);
+            }
+        } else if matches!(sender, OutgoingType::RemoveStake) {
+            value = Coin::from_u64_unchecked(2 * Policy::MINIMUM_STAKE)
+                .checked_sub(fee)
+                .expect("Fee must not be higher than staker stake");
             if fail_sender {
                 value += Coin::from_u64_unchecked(1);
             }
         }
+
         // Ensure that the value is the validator deposit if we are creating a validator.
+        // Ensure that the value is the minimum stake is respected in the staker operations.
         if matches!(recipient, IncomingType::CreateValidator) {
             value = Coin::from_u64_unchecked(Policy::VALIDATOR_DEPOSIT);
+        } else if matches!(recipient, IncomingType::CreateStaker) {
+            value = Coin::from_u64_unchecked(Policy::MINIMUM_STAKE);
+        } else if matches!(recipient, IncomingType::SetActiveStake) {
+            value = Coin::ZERO;
+        } else if matches!(recipient, IncomingType::RetireStake) {
+            value = Coin::from_u64_unchecked(Policy::MINIMUM_STAKE);
```

Comment:
This looks like it could benefit from an actual `match` instead of many `if matches!`.